RandLAPACK/misc/rl_util.hh

```diff
@@ -134,20 +134,53 @@ void col_swap(
     int64_t m,
     int64_t n,
     int64_t k,
-    std::vector<T> &A,
+    T* A,
+    int64_t lda,
     std::vector<int64_t> idx
 ) {
+    if(k > n) 
+        throw std::runtime_error("Incorrect rank parameter.");
 
-    if(k > std::min(m, n)) 
+    int64_t i, j; //, l;
+    for (i = 0, j = 0; i < k; ++i) {
+        j = idx[i] - 1;
+        blas::swap(m, &A[i * lda], 1, &A[j * lda], 1);
+
+        // swap idx array elements
+        // Find idx element with value i and assign it to j
+        auto it = std::find(idx.begin() + i, idx.begin() + k, i + 1);
+        idx[it - (idx.begin())] = j + 1;
+/*
+        // Simple version of the above
+        for(l = i; l < k; ++l) {
+            if(idx[l] == i + 1) {
+                    idx[l] = j + 1;
+                    printf("%d\n", l);
+                    break;
+            }
+        }
+        idx[i] = i + 1;
+*/
```

Comment:
This change should (probably) also be applied to the function defined below.

Reply:
Yeah, it is in my newest branch where i overhaul utilities. Fixed tho.